### PR TITLE
ref(hybrid-cloud): Updates integration proxy to pass proxy path as a header

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Stacktrace linking for gitlab is currently broken due to a bug with path encoding.

Gitlab expects likes in the format of `<base_url>/api/v4/<repo_name>/<url_encoded_file_path>`. Because our integration proxy passes this path as part of the proxy base URL, it gets decoded automatically and results in a signature verification eror. This is propagated up as a 404 to the gitlab client which also obfuscates the issue further.

This PR does 3 things:
1. Path is encoded as part of a new `X-SENTRY-SUBNET-PATH` and is no longer included as part of the base proxy URL.
2. Shortens the URL proxy API to `/api/0/integration-proxy`.
3. Changes the header validation error from a 404 to a 400 error.

### CODE REVIEW

File: src/sentry/shared_integrations/client/proxy.py [hunk 1]
```diff
@@ -190,12 +191,23 @@ def finalize_request(self, prepared_request: PreparedRequest) -> PreparedRequest
         # E.g. client.get("/chat.postMessage") -> proxy_path = 'chat.postMessage'
         proxy_path = trim_leading_slashes(prepared_request.url[len(base_url) :])
```
Comment (markstory) [inline src/sentry/shared_integrations/client/proxy.py:192]: Is this comment still accurate?

**Review by markstory — APPROVED** [review]
Looks good to me. Is the subnet-path header part of the request signature?

**Comment by GabeVillalobos** [comment]
> Looks good to me. Is the subnet-path header part of the request signature?

Yep, it should be part of the HMAC signature already since we explicitly take the path.

File: src/sentry/shared_integrations/client/proxy.py (hunk 1 above)
Comment (GabeVillalobos) [reply src/sentry/shared_integrations/client/proxy.py:192]: Mostly I think. The only nuance here is how this proxy_path is used, but we are relying on trimming the leading slashes.

**Pull request merged**